### ISSUE TITLE
When I import os and cv2 in code, nuitka complie success, then run the resulting executable, I get errors: UnboundLocalError: local variable 'os' referenced before assignment

### ISSUE THREAD

**Issue by Rcity** [Nuitka/Nuitka#2525]
- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version

```
1.8.5
Commercial: None
Python: 3.7.13 (default, Mar 28 2022, 08:03:21) [MSC v.1916 64 bit (AMD64)]
Flavor: Anaconda Python
Executable: D:\Application\Miniconda3\envs\common\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
Version C compiler: D:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\Hostx64\x64\cl.exe (cl 14.2).
```

- How did you install Nuitka and Python

  `pip install nuitka`

- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip freeze
```
absl-py==0.15.0
addict==2.4.0
aiohttp==3.8.4
aiosignal==1.3.1
altgraph==0.17.2
anyio==3.5.0
argon2-cffi @ file:///home/conda/feedstock_root/build_artifacts/argon2-cffi_1640817743617/work
argon2-cffi-bindings @ file:///D:/bld/argon2-cffi-bindings_1649500465986/work
astor==0.8.1
astunparse==1.6.3
async-generator==1.10
async-timeout==4.0.2
asynctest==0.13.0
atomicwrites==1.4.0
attrs @ file:///home/conda/feedstock_root/build_artifacts/attrs_1640799537051/work
auto-py-to-exe==2.23.1
Automat==20.2.0
autopep8 @ file:///home/conda/feedstock_root/build_artifacts/autopep8_1635267974115/work
Babel==2.9.1
backcall @ file:///home/conda/feedstock_root/build_artifacts/backcall_1592338393461/work
backports.functools-lru-cache @ file:///home/conda/feedstock_root/build_artifacts/backports.functools_lru_cache_1618230623929/work
baidu-aip==4.16.3
beautifulsoup4 @ file:///home/conda/feedstock_root/build_artifacts/beautifulsoup4_1649463573192/work
bleach @ file:///home/conda/feedstock_root/build_artifacts/bleach_1649361991009/work
blis==0.7.7
boto3 @ file:///opt/conda/conda-bld/boto3_1649078879353/work
botocore @ file:///opt/conda/conda-bld/botocore_1649076662316/work
bottle==0.12.23
bottle-websocket==0.2.9
brotlipy==0.7.0
bs4==0.0.1
cachetools==5.0.0
catalogue==2.0.7
certifi==2021.10.8
cffi @ file:///C:/ci_310/cffi_1642682485096/work
charset-normalizer==2.0.12
click==8.0.4
cnocr==2.2.2.2
cnstd==1.2.2
codecov==2.1.12
colorama @ file:///home/conda/feedstock_root/build_artifacts/colorama_1602866480661/work
coloredlogs==15.0.1
colorlogs==0.1
constantly==15.1.0
coverage==6.4.1
cryptography @ file:///C:/ci/cryptography_1639414690059/work
cssselect==1.1.0
cvxpy==1.2.1
cycler==0.11.0
cymem==2.0.6
Cython==0.29.28
dataclasses==0.6
-e git+https://github.com/lbin/DCNv2.git@41c9ebb3f7e9e563529bb5efc831c3922de72b08#egg=DCNv2
debugpy @ file:///D:/bld/debugpy_1649586515597/work
decorator @ file:///home/conda/feedstock_root/build_artifacts/decorator_1641555617451/work
defusedxml @ file:///home/conda/feedstock_root/build_artifacts/defusedxml_1615232257335/work
dill==0.3.5.1
docker-pycreds==0.4.0
easydict==1.9
easyocr==1.6.2
ecos==2.0.10
Eel==0.14.0
EMD-signal==1.2.3
entrypoints @ file:///home/conda/feedstock_root/build_artifacts/entrypoints_1643888246732/work
et-xmlfile==1.1.0
fastai==2.1.0
fastcore==1.3.2
fastdownload==0.0.5
fastjsonschema @ file:///home/conda/feedstock_root/build_artifacts/python-fastjsonschema_1641751198313/work/dist
fastprogress==1.0.2
filelock==3.6.0
flake8==4.0.1
Flask==2.1.1
Flask-Cors==3.0.10
flatbuffers==1.12
flit_core @ file:///home/conda/feedstock_root/build_artifacts/flit-core_1645629044586/work/source/flit_core
fonttools==4.31.2
frozenlist==1.3.3
fsspec==2023.1.0
future==0.18.2
gast==0.2.2
gdown==4.4.0
gensim==4.2.0
gevent==22.10.1
gevent-websocket==0.10.1
gitdb==4.0.9
GitPython==3.1.27
google-auth==2.6.6
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
graphviz==0.20.1
greenlet==1.1.3.post0
grpcio==1.32.0
h11==0.13.0
h5py==2.10.0
humanfriendly==10.0
hyperlink==21.0.0
idna @ file:///tmp/build/80754af9/idna_1637925883363/work
imageio==2.10.1
imgviz==1.5.0
importlib-metadata==6.0.0
importlib-resources @ file:///home/conda/feedstock_root/build_artifacts/importlib_resources_1648277295813/work
incremental==21.3.0
iniconfig==1.1.1
ipykernel @ file:///D:/bld/ipykernel_1649090924592/work/dist/ipykernel-6.12.1-py3-none-any.whl
ipython @ file:///D:/bld/ipython_1645889036858/work
ipython-genutils==0.2.0
ipywidgets==7.7.0
isort==5.10.1
itemadapter==0.5.0
itemloaders==1.0.4
itsdangerous==2.1.2
jaraco.classes==3.2.1
jaraco.collections==3.5.1
jaraco.context==4.1.1
jaraco.functools==3.5.0
jaraco.structures==2.1.0
jaraco.text==3.8.0
jaraco.ui==2.3.0
jaraco.windows==5.7.0
jedi @ file:///D:/bld/jedi_1649067454004/work
Jinja2 @ file:///home/conda/feedstock_root/build_artifacts/jinja2_1648299710939/work
jmespath @ file:///Users/ktietz/demo/mc3/conda-bld/jmespath_1630583964805/work
joblib==1.1.0
json5==0.9.6
jsonschema @ file:///home/conda/feedstock_root/build_artifacts/jsonschema-meta_1642000296051/work
jupyter==1.0.0
jupyter-client @ file:///home/conda/feedstock_root/build_artifacts/jupyter_client_1649340884673/work
jupyter-console==6.4.3
jupyter-contrib-core==0.3.3
jupyter-contrib-nbextensions==0.5.1
jupyter-core @ file:///D:/bld/jupyter_core_1645024488475/work
jupyter-highlight-selected-word==0.2.0
jupyter-latex-envs==1.4.6
jupyter-nbextensions-configurator==0.4.1
jupyter-server==1.16.0
jupyterlab==3.3.3
jupyterlab-language-pack-zh-CN==3.3.post2
jupyterlab-pygments @ file:///home/conda/feedstock_root/build_artifacts/jupyterlab_pygments_1649431904336/work
jupyterlab-server==2.12.0
jupyterlab-widgets==1.1.0
Keras==2.3.1
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.2
kiwisolver==1.4.2
kwarray==0.6.2
labelme==5.0.1
langcodes==3.3.0
lightning-utilities==0.7.1
llvmlite==0.38.0
loguru==0.6.0
lxml==4.8.0
Markdown==3.3.6
MarkupSafe @ file:///D:/bld/markupsafe_1648737728033/work
matplotlib==3.2.2
matplotlib-inline @ file:///home/conda/feedstock_root/build_artifacts/matplotlib-inline_1631080358261/work
mccabe==0.6.1
mistune @ file:///D:/bld/mistune_1635844820103/work
mkl-fft==1.3.1
mkl-random @ file:///C:/ci/mkl_random_1626186163140/work
mkl-service==2.4.0
mmcv==1.5.3
more-itertools==8.13.0
mpmath==1.2.1
multidict==6.0.4
multiprocess==0.70.13
murmurhash==1.0.6
natsort==8.1.0
nb-conda @ file:///D:/bld/nb_conda_1611345740130/work
nb-conda-kernels @ file:///D:/bld/nb_conda_kernels_1637000165884/work
nbclassic==0.3.7
nbclient @ file:///home/conda/feedstock_root/build_artifacts/nbclient_1646999386773/work
nbconvert @ file:///home/conda/feedstock_root/build_artifacts/nbconvert-meta_1648822144012/work
nbformat @ file:///home/conda/feedstock_root/build_artifacts/nbformat_1649105480291/work
nest-asyncio @ file:///home/conda/feedstock_root/build_artifacts/nest-asyncio_1648959695634/work
networkx==2.6.3
ninja==1.11.1
notebook @ file:///home/conda/feedstock_root/build_artifacts/notebook_1647377876077/work
notebook-shim==0.1.0
Nuitka==1.8.5
numba==0.55.1
numpy==1.21.6
oauthlib==3.2.0
onnx==1.13.1
onnxruntime==1.14.1
opencv-contrib-python==4.5.4.60
opencv-python==4.5.4.60
opencv-python-headless==4.5.4.60
openpyxl==3.0.10
opt-einsum==3.3.0
ordered-set==4.1.0
osqp==0.6.2.post5
outcome==1.1.0
packaging @ file:///home/conda/feedstock_root/build_artifacts/packaging_1637239678211/work
paddlepaddle==2.2.2
pandas==1.3.5
pandocfilters @ file:///home/conda/feedstock_root/build_artifacts/pandocfilters_1631603243851/work
parsel==1.6.0
parso @ file:///home/conda/feedstock_root/build_artifacts/parso_1638334955874/work
path==16.4.0
pathos==0.2.9
pathtools==0.1.2
pathy==0.6.1
patsy==0.5.3
pefile==2021.9.3
pickleshare @ file:///home/conda/feedstock_root/build_artifacts/pickleshare_1602536217715/work
Pillow==9.0.1
plotly==5.17.0
pluggy==1.0.0
Polygon3==3.0.9.1
pox==0.3.1
ppft==1.7.6.5
preshed==3.0.6
progress==1.6
prometheus-client @ file:///home/conda/feedstock_root/build_artifacts/prometheus_client_1649447152425/work
promise==2.3
prompt-toolkit @ file:///home/conda/feedstock_root/build_artifacts/prompt-toolkit_1649130487073/work
Protego==0.2.1
protobuf==3.20.3
psutil @ file:///D:/bld/psutil_1648873136396/work
py==1.11.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pyclipper==1.3.0.post4
pycocotools==2.0.4
pycodestyle @ file:///home/conda/feedstock_root/build_artifacts/pycodestyle_1633982426610/work
pycparser @ file:///tmp/build/80754af9/pycparser_1636541352034/work
pydantic==1.8.2
PyDispatcher==2.0.5
pyflakes==2.4.0
Pygments @ file:///home/conda/feedstock_root/build_artifacts/pygments_1641580240686/work
pyinstaller==4.10
pyinstaller-hooks-contrib==2022.3
pyOpenSSL @ file:///opt/conda/conda-bld/pyopenssl_1643788558760/work
pyparsing @ file:///home/conda/feedstock_root/build_artifacts/pyparsing_1649603503565/work
PyQt5==5.15.6
PyQt5-Qt5==5.15.2
PyQt5-sip==12.9.1
pyreadline==2.1
pyrsistent @ file:///D:/bld/pyrsistent_1649013563648/work
PySocks @ file:///C:/ci/pysocks_1594394709107/work
pytest==7.1.2
pytest-cov==3.0.0
pytest-runner==6.0.0
python-bidi==0.4.2
python-dateutil @ file:///tmp/build/80754af9/python-dateutil_1626374649649/work
python-docx==0.8.11
pytorch-lightning==1.9.3
pytz==2022.1
PyWavelets==1.3.0
pywin32==303
pywin32-ctypes==0.2.0
pywinpty==2.0.5
PyYAML==6.0
pyzmq @ file:///D:/bld/pyzmq_1649055808942/work
qdldl==0.1.5.post2
qtconsole==5.3.0
QtPy==2.0.1
queuelib==1.6.2
regex==2022.6.2
requests==2.27.1
requests-file==1.5.1
requests-oauthlib==1.3.1
rsa==4.8
s3transfer @ file:///tmp/build/80754af9/s3transfer_1626435152308/work
scikit-image==0.19.2
scikit-learn==1.0.2
scipy==1.7.3
Scrapy==2.6.1
scs==3.2.0
seaborn==0.11.2
selenium==4.1.3
Send2Trash @ file:///home/conda/feedstock_root/build_artifacts/send2trash_1628511208346/work
sentry-sdk==1.6.0
service-identity==21.1.0
setproctitle==1.2.3
Shapely==1.8.2
shortuuid==1.0.9
six==1.15.0
smart-open==5.2.1
smmap==5.0.0
sniffio==1.2.0
sortedcontainers==2.4.0
soupsieve @ file:///home/conda/feedstock_root/build_artifacts/soupsieve_1638550740809/work
spacy==3.2.4
spacy-legacy==3.0.9
spacy-loggers==1.0.2
srsly==2.4.2
statsmodels==0.13.5
sympy==1.10.1
tabulate==0.9.0
tenacity==8.2.3
tensorboard==1.15.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow==1.15.0
tensorflow-estimator==1.15.1
termcolor==1.1.0
terminado==0.13.3
terminaltables==3.1.10
testpath @ file:///home/conda/feedstock_root/build_artifacts/testpath_1645693042223/work
thinc==8.0.15
thop==0.1.0.post2207010342
threadpoolctl==3.1.0
tifffile==2021.11.2
tldextract==3.2.0
toml @ file:///home/conda/feedstock_root/build_artifacts/toml_1604308577558/work
tomli==2.0.1
torch==1.7.0+cpu
torchaudio==0.7.0
torchmetrics==0.11.1
torchvision==0.8.1+cpu
tornado @ file:///D:/bld/tornado_1648827404315/work
tqdm==4.64.0
traitlets @ file:///home/conda/feedstock_root/build_artifacts/traitlets_1635260543454/work
trio==0.20.0
trio-websocket==0.9.2
Twisted==22.2.0
twisted-iocpsupport==1.0.2
typer==0.4.1
typing-extensions==3.10.0.2
ubelt==1.1.1
Unidecode==1.3.6
urllib3 @ file:///opt/conda/conda-bld/urllib3_1643638302206/work
w3lib==1.22.0
wandb==0.12.21
wasabi==0.9.1
wcwidth @ file:///home/conda/feedstock_root/build_artifacts/wcwidth_1600965781394/work
webencodings==0.5.1
websocket-client==1.3.2
Werkzeug==2.1.1
wget==3.2
whichcraft==0.6.1
widgetsnbextension==3.6.0
win-inet-pton @ file:///C:/ci/win_inet_pton_1605306165655/work
win32-setctime==1.1.0
wincertstore==0.2
wrapt==1.12.1
wsproto==1.1.0
xdoctest==1.0.0
xgboost==1.6.2
xlrd==2.0.1
xlwt==1.3.0
xmltodict==0.12.0
yapf==0.32.0
yarl==1.8.2
zconst==1.0.0
zipp @ file:///home/conda/feedstock_root/build_artifacts/zipp_1649012893348/work
zope.event==4.5.0
zope.interface==5.4.0
zstandard==0.21.0
```

- Many times when you get an error from Nuitka, your setup may be special

 when without os and cv2 everthing be okay

- Also supply a Short, Self Contained, Correct, Example

  text of demo_all.py
```
import os
import cv2

def say(info):
    return "Say " + info

def main():
    print(say("Hello World"))

if __name__ == "__main__":
    main()
```
Compiling demo_all appears to work. When running the resulting executable, I get errors. Commands to repeat my simple test:
 ` python -m nuitka --standalone  --show-progress --show-memory --nofollow-imports ./demo_all.py `
 `demo_all.exe`

- Provide in your issue the Nuitka options used
 I got the issue:
 ```
(common) D:\inspur_rongyaru\code\nuitka_project\demo_all.dist>demo_all.exe
D:\inspur_rongyaru\code\nuitka_project\demo_all.dist\numpy\_distributor_init.py:30: UserWarning: loaded more than 1 DLL from .libs:
D:\inspur_rongyaru\code\nuitka_project\demo_all.dist\numpy\.libs\libopenblas.QVLO2T66WEPI7JZ63PS3HMOHFEY472BC.gfortran-win_amd64.dll
D:\inspur_rongyaru\code\nuitka_project\demo_all.dist\numpy\.libs\libopenblas.XWYDX2IKJW2NMTWSFYNGFUWKQU3LYTCZ.gfortran-win_amd64.dll
Traceback (most recent call last):
  File "D:\inspur_rongyaru\code\nuitka_project\demo_all.dist\demo_all.py", line 2, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "D:\inspur_rongyaru\code\nuitka_project\demo_all.dist\cv2\__init__.py", line 197, in <module cv2>
  File "D:\inspur_rongyaru\code\nuitka_project\demo_all.dist\cv2\__init__.py", line 93, in bootstrap
UnboundLocalError: local variable 'os' referenced before assignment
 ```

- Note if this is a regression
   Not working in any version.

**Comment by kayhayen** [comment]
I wonder what you expect `--nofollow-imports` to do. I suspect your report is incomplete, because it shouldn't even have cv2, but alas.
